src/state_machine_fuzz.zig

```diff
@@ -0,0 +1,151 @@
+
+    // With bits * 2, there's a ~50% chance of generating a uniform integer within the full range,
+    // and a ~50% chance of generating an integer biased towards an edge.
+    const bias_to = prng.range_inclusive(T, 0, bits * 2);
+
+    if (bias_to > bits) {
+        return prng.int(T);
+    } else {
+        const bias_center: T = if (bias_to == bits)
+            std.math.maxInt(T)
+        else
+            std.math.pow(T, 2, bias_to);
+        const bias_min = if (bias_to == 0) 0 else bias_center - @min(bias_center, 8);
+        const bias_max = if (bias_to == bits) bias_center else bias_center + 8;
+
+        return prng.range_inclusive(T, bias_min, bias_max);
+    }
+}
+
+pub fn main(allocator: std.mem.Allocator, args: fuzz.FuzzArgs) !void {
+    var context: TestContext = undefined;
+    try context.init(allocator);
+    defer context.deinit(allocator);
+
+    const request_buffer_full = try allocator.alignedAlloc(
+        u8,
+        16,
+        TestContext.message_body_size_max,
+    );
+    defer allocator.free(request_buffer_full);
+
+    const reply_buffer = try allocator.alignedAlloc(
+        u8,
+        16,
+        TestContext.message_body_size_max,
+    );
+    defer allocator.free(reply_buffer);
+
+    var prng = stdx.PRNG.from_seed(args.seed);
+
+    var op: u64 = 1;
+
+    for (0..args.events_max orelse 10_000_000) |_| {
+        const operation = prng.enum_uniform(TestContext.StateMachine.Operation);
+        const operation_size: u64 = switch (operation) {
+            inline else => |tag| @sizeOf(TestContext.StateMachine.EventType(tag)),
+        };
+        const request_buffer_operation = request_buffer_full[0..operation_size];
+
+        switch (operation) {
+            .query_accounts, .query_transfers => {
+                const query_filter: *tb.QueryFilter = @ptrCast(request_buffer_operation);
+                var reserved = std.mem.zeroes([6]u8);
+                if (prng.chance(.{ .numerator = 1, .denominator = 1000 })) {
+                    prng.fill(&reserved);
+                }
+
+                query_filter.* = .{
+                    .user_data_128 = int_edge_biased(&prng, u128),
+                    .user_data_64 = int_edge_biased(&prng, u64),
+                    .user_data_32 = int_edge_biased(&prng, u32),
+                    .ledger = int_edge_biased(&prng, u32),
+                    .code = int_edge_biased(&prng, u16),
+                    .timestamp_min = int_edge_biased(&prng, u64),
+                    .timestamp_max = int_edge_biased(&prng, u64),
+                    .limit = int_edge_biased(&prng, u32),
+                    .reserved = reserved,
+                    .flags = .{
+                        .reversed = prng.boolean(),
+                        .padding = if (prng.chance(.{ .numerator = 1, .denominator = 1000 }))
+                            int_edge_biased(&prng, u31)
+                        else
+                            0,
+                    },
```

Comment:
There are many `if (field != 0)` when we build the scan in `get_scan_from_account_filter` and `get_scan_from_query_filter`.

Do we have significant occurrences of "zeros" here to stress the combination of these code paths?

Reply:
Good question! So testing `int_edge_biased`, we get about ~15k zeroes from 10 million integers. or ~0.15% chance of hitting a zero.

We could increase that?

Reply:
Currently, it'll do 10m iterations in ~5 seconds, and it does 9 `int_edge_biased` calls for either `query_*` or `get_account_*`.

Reply:
> We could increase that?

Having zeros "sometime" is good enough 👍

Since we are not asserting semantically the results, we dont require testing the combination off all possible scans here (for this purpose we have the scan fuzzer and the vopr).